core/src/channel_group/mod.rs

```diff
@@ -60,26 +35,44 @@ impl ChannelGroup {
         let mut sample_cache_vecs = Vec::new();
 
         // Thread pool for individual channels to split between keys
-        let pool = if config.use_threadpool {
-            Some(Arc::new(rayon::ThreadPoolBuilder::new().build().unwrap()))
-        } else {
-            None
+        let channel_pool = match config.parallelism.channel {
+            ThreadCount::None => None,
+            ThreadCount::Auto => Some(Arc::new(rayon::ThreadPoolBuilder::new().build().unwrap())),
```

Comment:
It already uses rayon's automatic selection. For `ThreadCount::Auto` we don't specify a number of threads in rayon so it determines it itself. You mean to have `Auto` be a value we set ourselves for each operation?

Reply:
What I'm saying is, rayon's default/auto thread count implementation in the future would be changed to be dynamic instead of defaulting to your cpu thread count. The `num_threads` function's doc comment explains it iirc. Do we force it to be the thread count ourselves manually, for more predictable timings? Or do we keep rayon's future implementation so that it's more dynamic, e.g. not using threads it doesn't actually need for lower intensity use cases

Reply:
I am not very familiar with that but I think we could stick to rayon's default for `Auto`. And perhaps, when they implement it, if we see that a specific thread count works better for XSynth we could set the default value of the struct to that, but keep the auto value there.

Reply:
Yeah sounds good